dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -503,6 +504,35 @@ def _multi_svd_norm(x, row_axis, col_axis, op):
     return result
 
 
+def _nrm2_last_axis(x):
```

Comment:
is there any benefit to have a separate call instead of using `linalg.norm`?

Reply:
This improves performance rather than `return dpnp.square(dpnp.linalg.norm(x, axis=-1))`


```
def old_impl(x):
    real_dtype = _real_type(x.dtype)
    if dpnp.issubdtype(x.dtype, dpnp.complexfloating):
        return dpnp.sum(dpnp.abs(x) ** 2, axis=-1, dtype=real_dtype)
    else:
        return dpnp.sum(dpnp.square(x), axis=-1, dtype=real_dtype)

def new_impl(x):
    return dpnp.square(dpnp.linalg.norm(x, axis=-1))

```
```
In [1]: import numpy
In [2]: import dpnp
In [3]: from qwe import new_impl, old_impl

In [4]: na = numpy.random.randint(-10**4, 10**4, size=(1024,512))

In [5]: a_dp = dpnp.array(na,dtype='f4',device='cpu')

In [6]: %timeit run_old = old_impl(a_dp)
1.04 ms ± 25.5 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

In [7]: %timeit run_new = new_impl(a_dp)
2.15 ms ± 454 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

In [8]: a_dp = dpnp.array(na,dtype='f4',device='gpu')

In [9]: %timeit run_old = old_impl(a_dp)
394 µs ± 8.44 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

In [10]: %timeit run_new = new_impl(a_dp)
839 µs ± 6.49 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

In [11]: a_dp = dpnp.array(na,dtype='int64',device='cpu')

In [12]: %timeit run_old = old_impl(a_dp)
1.52 ms ± 302 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

In [13]: %timeit run_new = new_impl(a_dp)
4.5 ms ± 311 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [14]: a_dp = dpnp.array(na,dtype='int64',device='gpu')

In [15]: %timeit run_old = old_impl(a_dp)
1.05 ms ± 444 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [16]: %timeit run_new = new_impl(a_dp)
1.08 ms ± 52.4 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

```